### PR TITLE
Updating customconfig to string

### PR DESCRIPTION
**Pull request by q3blend**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

spec: https://github.com/Azure/azure-rest-api-specs/pull/3799

### CODE REVIEW

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Microsoft.Azure.CognitiveServices.Search.CustomSearch.csproj
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Microsoft.Azure.CognitiveServices.Search.CustomSearch.csproj]: Please bump the version and update PackageReleaseNotes here

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Generated/CustomSearch/CustomInstance.cs
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Generated/CustomSearch/CustomInstance.cs]: Changing parameter types could potentially be a breaking change, @shahabhijeet can you confirm

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Properties/AssemblyInfo.cs]: Please bump the assemblyversion and update packagereleasenotes in csproj

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt
Comment (dsgouda) [inline src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt]: Was this manually edited?
Was the REST spec PR merged?

**Review by dsgouda — CHANGES_REQUESTED** [review]
It looks like the REST spec is still not merged. We cannot merge this

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt
Comment (q3blend) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt]: It was generated with a hack (looking at a different repo) to avoid waiting until the spec PR is merged. The spec PR is now merged. I can regenerate if needed.

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Generated/CustomSearch/CustomInstance.cs
Comment (q3blend) [reply src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Generated/CustomSearch/CustomInstance.cs]: This is intended, we are bumping the SDK version

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt
Comment (dsgouda) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt]: Please regenerate and check in all artifacts generated.
Please avoid using hacks in the future,, if you wish to generate SDK using a branch that has not been merged yet, the `Start-AutoRestCodeGeneration` cmdlet has parameters to allow that

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Microsoft.Azure.CognitiveServices.Search.CustomSearch.csproj
Comment (dsgouda) [reply src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Microsoft.Azure.CognitiveServices.Search.CustomSearch.csproj]: Please look at an example [here](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Network/Management.Network/Microsoft.Azure.Management.Network.csproj#L13)

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomSearch.txt
Comment (dsgouda) [inline src/SDKs/_metadata/cognitiveservices_data-plane_CustomSearch.txt]: Please run `msbuild build.proj` from repo root and regenerate the code

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt
Comment (dsgouda) [inline src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt]: Are 2 .txt files being modified when you run the generate.ps1, please undo these changes and regenerate the code, there should be only 1 .txt file modified

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt
Comment (dsgouda) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomImageSearch.txt]: In that case, assembly versions and PackageReleaseNotes for both SDKs must be updated

File: src/SDKs/_metadata/cognitiveservices_data-plane_CustomSearch.txt
Comment (q3blend) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomSearch.txt]: Build started 9/16/2018 12:10:02 PM.
Project "C:\Users\ahosma\Workspace\azure-sdk-for-net\build.proj" on node 1 (default targets).
C:\Users\ahosma\Workspace\azure-sdk-for-net\tools\bootstrapTools\bootstrap.targets(5,4): error MSB4062: The "GetBuildTo
ols" task could not be loaded from the assembly C:\Users\ahosma\Workspace\azure-sdk-for-net\tools\bootstrapTools\taskBi
naries\Microsoft.Azure.Build.BootstrapTasks.dll. Could not load file or assembly 'Microsoft.Azure.Build.BootstrapTasks,
 Version=0.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. Strong name validation
failed. (Exception from HRESULT: 0x8013141A) Confirm that the <UsingTask> declaration is correct, that the assembly and
 all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framewor
k.ITask. [C:\Users\ahosma\Workspace\azure-sdk-for-net\build.proj]
Done Building Project "C:\Users\ahosma\Workspace\azure-sdk-for-net\build.proj" (default targets) -- FAILED.

I get the error above, I don't remember running msbuild, I always just generated the code using Autorest and that was enough. Why do I need to do this step?

Comment (dsgouda) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomSearch.txt]: Please disable strong name verification. (Sn -Vr) and try again

Comment (q3blend) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomSearch.txt]: @dsgouda I am getting a different error below
"C:\Users\ahosma\Workspace\azure-sdk-for-net\build.proj" (default target) (1) ->
(GetLatestBuildTools target) ->
  C:\Users\ahosma\Workspace\azure-sdk-for-net\tools\bootstrapTools\bootstrap.targets(5,4)
: error MSB4002: There was a failure retrieving the attributes for parameters in the "Get
BuildTools" task. Could not load type 'Microsoft.Build.Framework.IBuildEngine5' from asse
mbly 'Microsoft.Build.Framework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7
f11d50a3a'. [C:\Users\ahosma\Workspace\azure-sdk-for-net\build.proj]
  C:\Users\ahosma\Workspace\azure-sdk-for-net\tools\bootstrapTools\bootstrap.targets(5,4)
: error MSB4060: The "GetBuildTools" task has been declared or used incorrectly, or faile
d during construction. Check the spelling of the task name and the assembly name. [C:\Use
rs\ahosma\Workspace\azure-sdk-for-net\build.proj]

as I said before, why do I need to do this step? I've made several PRs before and I am sure I did not do msbuild

Comment (dsgouda) [reply src/SDKs/_metadata/cognitiveservices_data-plane_CustomSearch.txt]: @q3blend please ping offline and I shall resolve the issue

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Properties/AssemblyInfo.cs
Comment (dsgouda) [reply src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Properties/AssemblyInfo.cs]: The last published Nuget version was `1.0.0`, I think `AssemblyFileVersion` can be left as it is, please update the version in corresponding csproj

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Properties/AssemblyInfo.cs]: Please update package version in the corresponding csproj file

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomImageSearch/BingCustomImageSearch/Properties/AssemblyInfo.cs]: `AssemblyVersion` must remain the same, only the `AssemblyFileVersion` must be updated here

File: src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Properties/AssemblyInfo.cs
Comment (dsgouda) [inline src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Properties/AssemblyInfo.cs]: Same as above for `AssemblyVersion`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good, please address the 2 comments and squash your commits into a single commit, should be good to merge once addressed

**Comment by q3blend** [comment]
@dsgouda can you merge this? I believe all the comments were addressed

**Review by dsgouda — APPROVED** [review]

**Pull request merged**
